gapic-common/test/gapic/rest/firestore_server_stream_test.rb

```diff
@@ -0,0 +1,104 @@
+
+# frozen_string_literal: true
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "test_helper"
+require "faraday"
+require "pp"
+require 'pry'
+require "googleauth"
+
+#
+# Tests for the REST server stream.
+#
+class FirestoreServerStreamTest < Minitest::Test
+  def test_array_enumerable_stream
+    enumerable = [
+      "[",
+      "{",
+      "\"foo\":1",
+      "}",
+      ",",
+      "{",
+      "\"bar\":1",
+      "}",
+      "]"
+    ].to_enum
```

Comment:
For a test like this, consider something like

```ruby
"[{\"foo\":1},{\"bar\":1}]".chars
````

so that you test what happens when chunks boundaries fall anywhere, even in the middle of strings.

Reply:
Done.